includes/base/controls-stack.php

```diff
@@ -744,14 +746,32 @@ public function get_stack() {
 				$control['style_fields'] = $style_fields;
 			}
 
-			if ( ! empty( $control['selectors'] ) || ! empty( $control['dynamic'] ) || ! empty( $control['style_fields'] ) ) {
+			if ( ! empty( $control['selectors'] ) || ! empty( $control['dynamic'] ) || $this->is_global_control( $control_name, $controls ) || ! empty( $control['style_fields'] ) ) {
 				$style_controls[ $control_name ] = $control;
 			}
 		}
 
 		return $style_controls;
 	}
 
+	private function is_global_control( $control_name, $controls ) {
+		$control = $controls[ $control_name ];
+
+		$control_global_key = $control_name;
+
+		if ( ! empty( $control['groupType'] ) ) {
+			$control_global_key = $control['groupPrefix'] . $control['groupType'];
+		}
+
+		if ( empty( $controls[ $control_global_key ]['global']['active'] ) ) {
+			return false;
+		}
+
+		$globals = $this->get_settings( '__globals__' );
```

Comment:
What about default global?

Reply:
The default values are not printed by this mechanism